src/psyclone/tests/psyir/transformations/loop_fusion_test.py

```diff
@@ -0,0 +1,585 @@
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# ----------------------------------------------------------------------------
+# Author: J. Henrichs, Bureau of Meteorology
+# Modifies R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
+# Modified I. Kavcic, Met Office
+
+'''This module tests the loop fusion transformation.
+'''
+
+from __future__ import absolute_import, print_function
+
+import pytest
+
+from fparser.common.readfortran import FortranStringReader
+
+from psyclone.domain.nemo.transformations import NemoLoopFuseTrans
+from psyclone.psyGen import PSyFactory
+from psyclone.psyir.backend.fortran import FortranWriter
+from psyclone.psyir.nodes import Literal, Loop, Schedule, Return
+from psyclone.psyir.symbols import DataSymbol, INTEGER_TYPE
+from psyclone.psyir.transformations import LoopFuseTrans, TransformationError
+
+
+# ----------------------------------------------------------------------------
+def test_fusetrans_error_incomplete():
+    ''' Check that we reject attempts to fuse loops which are incomplete. '''
+    sch = Schedule()
+    loop1 = Loop(variable=DataSymbol("i", INTEGER_TYPE), parent=sch)
+    loop2 = Loop(variable=DataSymbol("j", INTEGER_TYPE), parent=sch)
+    sch.addchild(loop1)
+    sch.addchild(loop2)
+
+    fuse = LoopFuseTrans()
+
+    # Check first loop
+    with pytest.raises(TransformationError) as err:
+        fuse.validate(loop1, loop2)
+    assert ("Error in LoopFuseTrans transformation. The target loop must have "
+            "four children but found: []" in str(err.value))
+
+    loop1.addchild(Literal("start", INTEGER_TYPE, parent=loop1))
+    loop1.addchild(Literal("stop", INTEGER_TYPE, parent=loop1))
+    loop1.addchild(Literal("step", INTEGER_TYPE, parent=loop1))
+    loop1.addchild(Schedule(parent=loop1))
+    loop1.loop_body.addchild(Return(parent=loop1.loop_body))
+
+    # Check second loop
+    with pytest.raises(TransformationError) as err:
+        fuse.validate(loop1, loop2)
+    assert ("Error in LoopFuseTrans transformation. The target loop must have "
+            "four children but found: []" in str(err.value))
+
+    loop2.addchild(Literal("start", INTEGER_TYPE, parent=loop2))
+    loop2.addchild(Literal("stop", INTEGER_TYPE, parent=loop2))
+    loop2.addchild(Literal("step", INTEGER_TYPE, parent=loop2))
+    loop2.addchild(Schedule(parent=loop2))
+    loop2.loop_body.addchild(Return(parent=loop2.loop_body))
+
+    # Validation should now pass
+    fuse.validate(loop1, loop2)
+
+
+# ----------------------------------------------------------------------------
+def test_fusetrans_error_not_same_parent():
+    ''' Check that we reject attempts to fuse loops which don't share the
+    same parent '''
+
+    sch1 = Schedule()
+    sch2 = Schedule()
+    loop1 = Loop(variable=DataSymbol("i", INTEGER_TYPE), parent=sch1)
```

Comment:
I appreciate that you've just moved this test over but it could be shortened a lot by using `loop1 = Loop.create(DataSymbol("i", INTEGER_TYPE), Literal("1", INTEGER_TYPE), Literal("10", INTEGER_TYPE), Literal("1", INTEGER_TYPE), [Return()])`

Reply:
Done.